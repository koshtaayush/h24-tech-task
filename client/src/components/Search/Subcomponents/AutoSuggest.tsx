//Library
import React from 'react'
import styled from 'styled-components'

//Shared Interface
import { Article } from '../../../typings/types'

interface Props {
    articleList: Array<Article>
}

export const AutoSuggest: React.FC<Props> = (props) => {
    const { articleList } = props
    
    return (
        <Container>
            <SuggestionList>
                {articleList.map((article, index) => {
                    return (
                        <SuggestionListItem key={index}>{article.name}</SuggestionListItem>
                    )
                })}
            </SuggestionList>
        </Container>
    )
}

export default AutoSuggest

const Container = styled.div`
    width: 31.375rem;
    z-index: 9;
    position: absolute;
    display: block;
    -webkit-box-shadow: 0 0.0625rem 0.375rem transparent;
    box-shadow: 0 0.0625rem 0.375rem transparent;
    font-size: 0.8125rem;
    color: #696e79;
    background-color: #fff;
    -webkit-font-smoothing: antialiased;
    -webkit-box-shadow: 0 0.25rem 0.75rem 0 rgba(0, 0, 0, 0.05);
    box-shadow: 0 0.25rem 0.75rem 0 rgba(0, 0, 0, 0.05);
`

const SuggestionList = styled.ul`
    margin: 0 0 0.625rem;
    padding: 0;
`

const SuggestionListItem = styled.li`
    list-style: none;
    padding: 0.9375rem 0.25rem 0.9375rem 0.75rem;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
`
