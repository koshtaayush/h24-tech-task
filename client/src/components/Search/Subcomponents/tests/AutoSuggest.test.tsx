import React from 'react'
import { shallow } from 'enzyme'

import AutoSuggest from './../AutoSuggest'

describe('AutoSuggest', () => {
    it('Should match the snapshot', () => {
        const articleList = [
            {
                name: 'Premium Komfortmatratze Smood',
                variantName: '180 x 200cm',
                prices: {
                    currency: 'EUR',
                    regular: {
                        value: 56999,
                    },
                },
                images: [
                    {
                        path:
                            'https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp',
                    },
                ],
            },
            {
                name: 'Premium Komfortmatratze Smood',
                variantName: '180 x 200cm',
                prices: {
                    currency: 'EUR',
                    regular: {
                        value: 56999,
                    },
                },
                images: [
                    {
                        path:
                            'https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp',
                    },
                ],
            },
        ]

        const component = shallow(<AutoSuggest articleList={articleList} />)

        expect(component).toMatchSnapshot()
    })
})
