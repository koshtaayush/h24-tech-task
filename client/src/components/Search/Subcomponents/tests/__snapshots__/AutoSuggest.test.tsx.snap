// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoSuggest Should match the snapshot 1`] = `
<styled.div>
  <styled.ul>
    <styled.li
      key="0"
    >
      Premium Komfortmatratze Smood
    </styled.li>
    <styled.li
      key="1"
    >
      Premium Komfortmatratze Smood
    </styled.li>
  </styled.ul>
</styled.div>
`;
